featurebyte/routes/feature_namespace/controller.py

```diff
@@ -164,6 +164,22 @@ async def update_feature_namespace(
                     "Cannot set default feature ID when default version mode is not MANUAL."
                 )
 
+            # check new default feature ID exists & make sure it is the highest readiness level among all versions
+            new_default_feature = await self.feature_service.get_document(
+                document_id=data.default_feature_id
+            )
+            max_readiness = new_default_feature.readiness
+            async for feature_dict in self.feature_service.list_documents_iterator(
+                query_filter={"_id": {"$in": feature_namespace.feature_ids}}
+            ):
+                max_readiness = max(max_readiness, FeatureReadiness(feature_dict["readiness"]))
+
+            if new_default_feature.readiness != max_readiness:
+                raise DocumentUpdateError(
+                    f"Cannot set default feature ID to {new_default_feature.id} "
+                    f"because it has lower readiness level than {max_readiness}."
```

Comment:
Would it be useful to include the version name(s) associated with the highest readiness in the message?

Reply:
Any recommended message?

Reply:
👌  Will update the message here as per request.